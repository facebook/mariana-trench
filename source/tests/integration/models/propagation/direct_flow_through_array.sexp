(method (private) "LFlow;.flow:()V"
 (
  LINE(1)
  (const v0 10)
  (new-array v0 "[LData;") ; create an array of length 10
  (move-result-pseudo-object v1)

  LINE(2)
  (invoke-virtual (v0) "LSource;.source:()LData;")
  (move-result-object v2)

  LINE(3)
  ; v1[0] = v2
  (const v0 0)
  (aput v2 v1 v0)

  LINE(4)
  ; v3 = read v1[0]
  (const v0 0)
  (aget v1 v0)
  (move-result-pseudo v3)

  LINE(5)
  (invoke-virtual (v0 v3) "LSink;.sink:(LData;)V")

  LINE(6)
  (return-void)
 )
)
