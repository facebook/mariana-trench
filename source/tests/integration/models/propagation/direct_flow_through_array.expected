{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "issues" :
      [
        {
          "always_features" :
          [
            "via-array"
          ],
          "position" :
          {
            "line" : 5,
            "path" : "Flow.java"
          },
          "rule" : 1,
          "sinks" :
          [
            {
              "call_position" :
              {
                "line" : 5,
                "path" : "Flow.java"
              },
              "callee" : "LSink;.sink:(LData;)V",
              "callee_port" : "Argument(1)",
              "distance" : 1,
              "kind" : "TestSink",
              "origins" :
              [
                "LSink;.sink:(LData;)V"
              ]
            }
          ],
          "sources" :
          [
            {
              "always_features" :
              [
                "via-array"
              ],
              "call_position" :
              {
                "line" : 2,
                "path" : "Flow.java"
              },
              "callee" : "LSource;.source:()LData;",
              "callee_port" : "Return",
              "distance" : 1,
              "kind" : "TestSource",
              "local_features" :
              {
                "always_features" :
                [
                  "via-array"
                ]
              },
              "local_positions" :
              [
                {
                  "line" : 3
                }
              ],
              "origins" :
              [
                "LSource;.source:()LData;"
              ]
            }
          ]
        }
      ],
      "method" : "LFlow;.flow:()V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      }
    }
  ]
}
